nextjs-13/src/middleware.ts

```diff
@@ -13,7 +13,11 @@ const middleware = async (req) => {
     return NextResponse.redirect(new URL('/', req.url));
   }
 
-  if (user === null && !url.startsWith('/auth')) {
+  if (
+    user === null &&
+    !url.startsWith('/auth') &&
+    !url.startsWith('/newsletter')
```

Comment:
When having more than two conditions checks, it is best to move it to a function:

```javascript
const isPublicPath(URL) {
  return url.startsWith('/newsletter') || url.startsWith('/auth');
}

const middleware = async (req) => {

if (!user && !isPublicPath(url)) {}
```

Reply:
Resolved [here](https://github.com/nimblehq/js-framework-benchmark/pull/127/commits)